Library/Homebrew/cmd/pull.rb

```diff
@@ -113,21 +113,23 @@ def pull
           "git", "diff-tree", "-r", "--name-only",
           "--diff-filter=AM", orig_revision, "HEAD", "--", tap.formula_dir.to_s
         ).each_line do |line|
+          next unless line.end_with? ".rb\n"
           name = "#{tap.name}/#{File.basename(line.chomp, ".rb")}"
-          begin
-            changed_formulae_names << name
-          # Make sure we catch syntax errors.
-          rescue Exception
-            next
-          end
+          changed_formulae_names << name
         end
       end
 
       fetch_bottles = false
       changed_formulae_names.each do |name|
         next if ENV["HOMEBREW_DISABLE_LOAD_FORMULA"]
 
-        f = Formula[name]
+        begin
+          f = Formula[name]
+        # Make sure we catch syntax errors.
+        rescue Exception
+          next
+        end
```

Comment:
@UniqMartin Is this what you had in mind? If I’m not mistaken it’s the only place we could get a syntax error.